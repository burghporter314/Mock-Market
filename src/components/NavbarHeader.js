import React from 'react';
import logo from '../assets/market.png'
import { Button, Navbar, Container, NavDropdown, Nav } from 'react-bootstrap';
import { useAuth0 } from '@auth0/auth0-react';

const NavbarHeader = () => {
    const { logout, user } = useAuth0();

    let rightNavbar;
    return(
    <Navbar bg="light" expand="lg">
    <Container>
        <Nav>
            <a href="/main">
                <img className="col-lg-6 img-fluid w-100 login-image center-block" src={logo} style={{"maxWidth": "40px"}}></img>
            </a>
        </Nav>
        <Navbar.Brand href="#home">Mock Market</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="me-auto">
                <Nav.Link href="/main">Dashboard</Nav.Link>
                <Nav.Link href="/search">Search</Nav.Link>
            </Nav>
            {user &&
                <React.Fragment>
                    <Navbar.Text style={{margin: "0 10px"}}>
                        Welcome, <a>{user.name}!</a>
                    </Navbar.Text>
                    <Nav>
                        <Button variant="danger" onClick={() => logout()}>Logout</Button>
                    </Nav>
                </React.Fragment>
            }
        </Navbar.Collapse>
    </Container>
    </Navbar>
    );
}

export default NavbarHeader;