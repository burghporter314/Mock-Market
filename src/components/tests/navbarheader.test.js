import React from 'react';
import NavbarHeader from '../NavbarHeader';
import renderer from 'react-test-renderer';

it('should match snapshot', () => {
    const component = renderer.create(
        <NavbarHeader/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})