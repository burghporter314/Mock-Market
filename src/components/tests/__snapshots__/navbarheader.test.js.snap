// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        href="/main"
      >
        <img
          className="col-lg-6 img-fluid w-100 login-image center-block"
          src="market.png"
          style={
            Object {
              "maxWidth": "40px",
            }
          }
        />
      </a>
    </div>
    <a
      className="navbar-brand"
      href="/main"
    >
      Mock Market
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="me-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="/main"
          disabled={false}
          href="/main"
          onClick={[Function]}
        >
          Dashboard
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/search"
          disabled={false}
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/transfer"
          disabled={false}
          href="/transfer"
          onClick={[Function]}
        >
          Transfer
        </a>
      </div>
    </div>
  </div>
</nav>
`;
